modules/kafka/kafka-dest-driver.c

```diff
@@ -281,25 +281,16 @@ _kafka_delivery_report_cb(rd_kafka_t *rk,
                           void *opaque, void *msg_opaque)
 {
   KafkaDestDriver *self = (KafkaDestDriver *) opaque;
-  LogMessage *msg = (LogMessage *) msg_opaque;
-
-  /* we already ACKed back this message to syslog-ng, it was kept in
-   * librdkafka queues so far but successfully delivered, let's unref it */
 
   if (err != RD_KAFKA_RESP_ERR_NO_ERROR)
     {
-      LogThreadedDestWorker *worker = (LogThreadedDestWorker *) self->super.workers[0];
-      LogQueue *queue = worker->queue;
-      LogPathOptions path_options = LOG_PATH_OPTIONS_INIT_NOACK;
-
-      msg_debug("kafka: delivery report for message came back with an error, putting it back to our queue",
+      msg_debug("kafka: delivery report for message came back with an error, message is lost",
```

Comment:
I've recently found that some error/warning message still indicate that on failure to send a message it will be readded to syslog-ng's queue:
https://github.com/syslog-ng/syslog-ng/blob/3e62aa98fafc1507f7fe8344b850e7e608e92c02/modules/kafka/kafka-dest-driver.c#L423-L426

We should change this to `... the remaining entries will be purged and lost`

Reply:
This should be fixed, also a commit close by.